src/linker/Linker.Steps/MarkStep.cs

```diff
@@ -3739,6 +3650,99 @@ public override bool HandleCall (MethodBody callingMethodBody, MethodReference c
 								}
 							}
 							break;
+
+						//
+						// System.Activator
+						// 
+						// static CreateInstance (string assemblyName, string typeName)
+						// static CreateInstance (string assemblyName, string typeName, bool ignoreCase, System.Reflection.BindingFlags bindingAttr, System.Reflection.Binder? binder, object?[]? args, System.Globalization.CultureInfo? culture, object?[]? activationAttributes)
+						// static CreateInstance (string assemblyName, string typeName, object?[]? activationAttributes)
+						//
+						case "CreateInstance" when !calledMethod.ContainsGenericParameter
+							&& calledMethod.DeclaringType.Namespace == "System"
+							&& calledMethod.DeclaringType.Name == "Activator"
+							&& calledMethod.Parameters.Count >= 2
+							&& calledMethod.Parameters [0].ParameterType.MetadataType == MetadataType.String
+							&& calledMethod.Parameters [1].ParameterType.MetadataType == MetadataType.String:
+							ProcessCreateInstanceByName (ref reflectionContext, calledMethodDefinition, methodParams);
+							break;
+
+						//
+						// System.Activator
+						// 
+						// static CreateInstanceFrom (string assemblyFile, string typeName)
+						// static CreateInstanceFrom (string assemblyFile, string typeName, bool ignoreCase, System.Reflection.BindingFlags bindingAttr, System.Reflection.Binder? binder, object? []? args, System.Globalization.CultureInfo? culture, object? []? activationAttributes)
+						// static CreateInstanceFrom (string assemblyFile, string typeName, object? []? activationAttributes)
+						//
+						case "CreateInstanceFrom" when !calledMethod.ContainsGenericParameter
+							&& calledMethod.DeclaringType.Namespace == "System"
+							&& calledMethod.DeclaringType.Name == "Activator"
+							&& calledMethod.Parameters.Count >= 2
+							&& calledMethod.Parameters [0].ParameterType.MetadataType == MetadataType.String
+							&& calledMethod.Parameters [1].ParameterType.MetadataType == MetadataType.String:
+							ProcessCreateInstanceByName (ref reflectionContext, calledMethodDefinition, methodParams);
+							break;
+
+						//
+						// System.Activator
+						// 
+						// static T CreateInstance<T> ()
+						//
+						case "CreateInstance" when calledMethod.ContainsGenericParameter
+							&& calledMethod.DeclaringType.Namespace == "System"
+							&& calledMethod.DeclaringType.Name == "Activator"
+							&& calledMethod.Parameters.Count == 0: {
+								// Not yet supported in any combination
+								reflectionContext.AnalyzingPattern ();
+								reflectionContext.RecordUnrecognizedPattern ($"Activator call '{reflectionContext.MethodCalled.FullName}' inside '{reflectionContext.MethodCalling.FullName}' is not supported");
```

Comment:
This might cause a lot of noise since this is what `new T()` in C# compiles to.

Would it make sense to at least check whether the generic argument has the `new()` constraint (in which case linker already made sure this works).

Later we can be more creative and e.g. check whether the T is a concrete type (in which case we can just keep the ctor here), or whether the T is a generic parameter that has a dataflow annotation that keeps the ctor (we don't have that yet).

Reply:
I added handling of the generic argument with constraint.